stdlib/strings/strings.go

```diff
@@ -487,6 +488,7 @@ func init() {
 				return values.NewString(strings.Join(newStringArray, argVals[1].Str())), nil
 			}, false,
 		),
+		"substring": substring,
```

Comment:
I think this map is for "special" functions that can't have their builtin bodies "generated" like with `generateDualArgStringFunctionReturnBool` for example.

It seems like `substring` also falls into this category but is being registered in a different way. 

Maybe the intent was to do something like 
```go
for k, v := range specialFns {
  runtime.RegisterPackageValue("strings",k, v)
}
```
In any case it looks like some rot has seeped in here and it might be good to make this consistent?

Reply:
The function is already registered, yes.

Best I could tell the map of "special" functions exists today solely for the purpose of exposing them to the test module. I didn't see any other access to the map outside of the now 2 fns, inserted here and the test module.

If you have a recommendation for how to get access to the function in the test module without this step, I'd be happy to remove it. It wasn't clear to me what API I should use to get a handle on the function otherwise.

Reply:
Actually, now that I know the repro uses NBSP instead of null bytes, I can try and replace the Go test with an acceptance test (which sidesteps the access issue for me) :sweat_smile:

I really struggled to get a null byte into the source for a test or in the repl, but in retrospect I might have succeeded many times. The null byte just wasn't contributing to the bug as originally thought.

**Update**: this seems to have worked. I'll push a new pair of revs to move the test and remove the map usage.